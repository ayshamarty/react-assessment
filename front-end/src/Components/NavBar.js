import React, { Component } from "react";
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import "../App.css";
import axios from "axios";

class NavBar extends Component {
      constructor() {
    super();
    this.state = {
      recipes: [],
      recipeID: ""
    };
  }
}

export default NavBar;