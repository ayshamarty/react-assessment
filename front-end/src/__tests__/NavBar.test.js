import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import NavBar from "../Components/NavBar";

import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<NavBar />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders shallow without crashing", () => {
  shallow(<NavBar />);
});

it('snapshot NavBar', () => {
  const wrapper = renderer.create(<NavBar/>).toJSON()
  expect(wrapper).toMatchSnapshot();
})

