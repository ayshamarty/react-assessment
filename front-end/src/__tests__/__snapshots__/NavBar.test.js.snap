// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot NavBar 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md navbar-light bg-dark"
  >
    <button
      aria-label={null}
      className="btn btn-secondary"
      id="user-button"
      onClick={[Function]}
    >
      <a
        className="link"
        href="/users"
        onClick={[Function]}
      >
        Users
      </a>
    </button>
    <button
      aria-label={null}
      className="btn btn-secondary"
      id="login-button"
      onClick={[Function]}
    >
      <a
        className="link"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </button>
    <button
      aria-label={null}
      className="btn btn-secondary"
      id="signup-button"
      onClick={[Function]}
    >
      <a
        className="link"
        href="/signup"
        onClick={[Function]}
      >
        Sign Up
      </a>
    </button>
  </nav>
  <br />
</div>
`;
